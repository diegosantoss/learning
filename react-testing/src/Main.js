import React from 'react';

const Main = ({ title, description}) => (
  <>
    <h1>{title}</h1>
    <p>{description}</p>
  </>
)

export default Main;