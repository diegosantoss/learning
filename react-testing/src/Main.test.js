import React from 'react';
import Main from './Main';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

it("should render Main component", () => {
  const props = {
    title: 'Titulo da p√°gina',
    description: 'Aqui um breve resumo de test'
  }

  const main = mount(<Main {...props} />);
  expect(toJson(main)).toMatchSnapshot();
})

it("should render Main component empty", () => {
  const props = {
    title: null,
    description: null
  }

  const main = mount(<Main {...props} />);
  expect(toJson(main)).toMatchSnapshot();
})

// it("should render the homepage for the non logged in user", () => {
//   const props = {
//     isLoggedIn: false,
//     username: null
//   }
//   const app = mount(<App {...props} />)
//   expect(toJson(app)).toMatchSnapshot()
// })