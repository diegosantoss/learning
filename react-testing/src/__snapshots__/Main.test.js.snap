// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Main component 1`] = `
<Main
  description="Aqui um breve resumo de test"
  title="Titulo da página"
>
  <h1>
    Titulo da página
  </h1>
  <p>
    Aqui um breve resumo de test
  </p>
</Main>
`;

exports[`should render Main component empty 1`] = `
<Main
  description={null}
  title={null}
>
  <h1 />
  <p />
</Main>
`;
